hazelcast/src/test/java/com/hazelcast/concurrent/lock/LockLeaseMemberBounceTest.java

```diff
@@ -0,0 +1,136 @@
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.HashSet;
+import java.util.Random;
+
+import static com.hazelcast.test.HazelcastTestSupport.assertTrueEventually;
+import static com.hazelcast.test.HazelcastTestSupport.generateKeyNotOwnedBy;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+/**
+ * Runs a cluster of {@value MEMBER_COUNT + 1} and calls lease lock from {@value DRIVER_COUNT} drivers. During the test bounces
+ * {@value MEMBER_COUNT} cluster members (shuts down and restarts). The remaining 1 member is stable (never bounced) but no
+ * locks are owned by this member.
+ * Asserts that all locks are unlocked eventually.
+ */
+@RunWith(HazelcastSerialClassRunner.class)
+@Category(SlowTest.class)
+public class LockLeaseMemberBounceTest {
+    private static final int MEMBER_COUNT = 4;
+    private static final int DRIVER_COUNT = 4;
+    private static final int LEASE_DURATION_MILLIS = 100;
+    private static final int TEST_DURATION_SECONDS = 40;
+    private static final int LOCK_COUNT = 500;
+    private static final int CONCURRENCY = 2;
+
+    @Rule
+    public BounceMemberRule bounceMemberRule = BounceMemberRule.with(getConfig())
+                                                               .clusterSize(MEMBER_COUNT)
+                                                               .driverCount(DRIVER_COUNT).build();
+    @Rule
+    public JitterRule jitterRule = new JitterRule();
+
+    @Before
+    public void setup() {
+        // Needed because BounceMemberRule expects a before method or otherwise won't bounce, will be fixed
+    }
+
+    @Test
+    public void leaseShouldExpireWhenMemberBouncing() {
+        final HazelcastInstance steadyMember = bounceMemberRule.getSteadyMember();
+
+        String[] lockNames = new String[LOCK_COUNT];
+        for (int i = 0; i < lockNames.length; i++) {
+            lockNames[i] = generateKeyNotOwnedBy(steadyMember);
+        }
+
+        final Driver[] drivers = new Driver[DRIVER_COUNT];
+        for (int i = 0; i < drivers.length; i++) {
+            drivers[i] = new Driver(bounceMemberRule.getNextTestDriver(), lockNames);
+        }
+
+        DriverRunnable[] runnables = new DriverRunnable[drivers.length * CONCURRENCY];
+        for (int i = 0; i < drivers.length; i++) {
+            for (int j = 0; j < CONCURRENCY; j++) {
+                runnables[i * CONCURRENCY + j] = new DriverRunnable(drivers[i]);
+            }
+        }
+
+        bounceMemberRule.testRepeatedly(runnables, TEST_DURATION_SECONDS);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                for (Driver driver : drivers) {
+                    driver.doAssert();
+                }
+            }
+        }, MILLISECONDS.toSeconds(LEASE_DURATION_MILLIS) + 10);
```

Comment:
I think timeout here is too small, it can cause accidental failures. Also note that, `assertTrueEventually(...)` runs `AssertTask` in 200ms intervals.